### PR TITLE
Implement PermissionDescription API

### PR DESCRIPTION
**Pull request by ST-DDT**
Adds https://github.com/SpongePowered/SpongeAPI/issues/601 (low priority)
Implements https://github.com/SpongePowered/SpongeAPI/pull/747 (description + discussion)

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "SpongeAPI"]
 	path = SpongeAPI
-	url = https://github.com/SpongePowered/SpongeAPI.git
+	url = https://github.com/ST-DDT/SpongeAPI.git
```
Comment (ryantheleach) [inline .gitmodules]: This probably shouldn't be included in the PR.

Comment (ST-DDT) [inline .gitmodules]: Otherwise the automated PR build/test support does not work.

Comment (kenzierocks) [inline .gitmodules]: To make it easier you can do it like I did with my Selectors PR, I kept the code in one commit and the API ref/module changes in another commit. This way the team can manipulate it easier.

Comment (ST-DDT) [inline .gitmodules]: Good advice. I'll change this later today.
